ui/pages/confirmations/confirmation/confirmation.js

```diff
@@ -514,7 +514,16 @@ export default function ConfirmationPage({
           </Box>
         ) : null}
         {useSnapHeader && (
-          <SnapAuthorshipHeader snapId={pendingConfirmation?.origin} />
+          <SnapAuthorshipHeader
+            snapId={pendingConfirmation?.origin}
+            onCancel={
+              templatedValues.onCancel ||
+              // /!\ Treat cancel as submit only if approval type is appropriate /!\
+              (pendingConfirmation?.type === ApprovalType.SnapDialogAlert
```

Comment:
Rather than adding further coupling to specific types in this generic component, could we instead just specify the `onReject` in the relevant templates such as `ui/pages/confirmations/confirmation/templates/snaps/snap-alert/snap-alert.js`?

Reply:
The problem with this is when `onCancel` is added to the `snap-alert.js`, the confirmation popup renders an extra button. Alert needs to have only one button by design. We just want to have the possibility to close popup from header everywhere (in all Snaps dialogs) and this case was somehow exception because there is no reason to have two buttons on alert screen.

Do you see quick fix for this maybe?

![Screenshot 2024-07-18 at 14 45 21](https://github.com/user-attachments/assets/e691850d-c9ff-4954-b701-4c583af968ff)

Reply:
Might be extra scope for now, but we could add a `hideCancelButton` property to the template to the  to match `hideSubmitButton` so you could still define an `onCancel` but explicitly hide the button so it's only used for this header exit behaviour.